### ISSUE TITLE
a question about data process

### ISSUE THREAD

**Issue by lazyloafer** [wangjlgz/Hypergraph-Session-Recommendation#2]
could you clearly state how you process the data for training and evaluation? You mentioned that you did data augmentation but what I am concerned about is whether the longer sequence would leak the information to the short sequence. For example, in sequence ([i_s1, i_s2], i_s3), it implies that we know i_s1 and i_s2 has relations in the graph. But for ([i_s1], i_s2), we suppose that we don't know i_s2 and i_s1 have a connection. So a longer sequence may contain the graph info of the short session's label. This is important we consider the problem from a graph view so I encourage the authors to provide details about how you generate the datasets, especially how you generate validation/testing set to avoid any potential data leak issue.
Looking forward to your reply！

**Comment by wangjlgz** [comment]
As we indicated in the paper, we followed the same data processing procedures as in [[16](https://arxiv.org/pdf/1711.04725.pdf),[19](https://dl.acm.org/doi/10.1145/3219819.3219950),[32](https://arxiv.org/abs/1811.00855)]. And you may also refer to [preprocess.py](https://github.com/CRIPAC-DIG/SR-GNN/blob/master/datasets/preprocess.py) provided by [[32](https://arxiv.org/abs/1811.00855)] for the detailed implementation. Our train/valid/test truncation is the same as in previous work.

The sessions for training and the sessions for testing are from different time periods. For example, for Diginetica, we train the model with the sessions happening in the early time periods (before the last 7-days), while testing the model with sessions happening in the last 7-days. This is what we do to simulate the training/testing process in the real-world scenario (i.e., train the model with data before a certain timestamp, and deploy it after that timestamp for testing) and avoid the data leakage. 

We understand that the existence of both sequence ([i_s1, i_s2], i_s3) and sequence ([i_s1], i_s2) in the training dataset could potentially lead to over-fitting. However, it doesn't leak any information to the testing phase. This is what the related literatures in session-based recommendation have done to get more training sessions by preprocessing the data for training phase only.
